buildSrc/src/main/kotlin/com/jakewharton/rxbinding/project/KotlinGenTask.kt

```diff
@@ -86,32 +104,48 @@ open class KotlinGenTask : SourceTask() {
     }
 
     private fun resolveTypeArguments(inputType: ClassOrInterfaceType,
-        methodAnnotations: List<AnnotationExpr>?): String {
+        methodAnnotations: List<AnnotationExpr>?,
+        associatedImports: Map<String, ClassName>? = null): List<TypeName> {
       return inputType.typeArgs?.map { type: Type ->
-          resolveKotlinType(type, methodAnnotations)
-        }?.joinToString(prefix = "<", postfix = ">") ?: ""
+        resolveKotlinType(type, methodAnnotations, associatedImports)
+      } ?: emptyList()
     }
 
     private fun resolveKotlinWildcardType(inputType: WildcardType,
-        methodAnnotations: List<AnnotationExpr>?): String {
-      val nullable = resolveNullableKotlinWildcardSuffix(methodAnnotations)
-      return if (inputType.`super` != null) {
-        "in ${resolveKotlinType(inputType.`super`)}$nullable"
-      } else if (inputType.extends != null) {
-        "out ${resolveKotlinType(inputType.extends)}$nullable"
-      } else {
-        throw IllegalStateException("Wildcard with no super or extends")
+        methodAnnotations: List<AnnotationExpr>?,
+        associatedImports: Map<String, ClassName>? = null): WildcardTypeName {
+      val isNullable = isWildcardNullable(methodAnnotations)
+      return when {
+        inputType.`super` != null -> WildcardTypeName.supertypeOf(
+            resolveKotlinType(inputType.`super`, associatedImports = associatedImports))
+            .let {
+              if (isNullable) {
+                it.asNullable()
+              } else {
+                it
+              }
+            }
```

Comment:
This is very vertically wasteful. Maybe we can clean it up to something like:

```kotlin
val isNullable = isWildcardNullable(methodAnnotations)
inputType.`super`?.let { name ->
  val type = WildcardTypeName.supertypeOf(resolveKotlinType(name), associatedImports = associatedImports)
  return if (isNullable) type.asNullable() else type
}
inputType.extends?.let { name ->
  // same thing
}
throw ISE(...)
```

Reply:
8e7cbf4